### PR TITLE
Unarchive timeout parameter and RETURN documentation.

### LINKED ISSUE #67445: unarchive module: missing documentation on return values
##### SUMMARY
The unarchive module is missing documentation on its return values.
Only the `list_files` option mentions it at all:
> If set to True, return the list of files that are contained in the tarball.

It does not mention how this list is returned (Spoiler: It's in the `files` key, as a list of file names).
A `RETURN` section should be added to the module documentation.
Ideally, the `EXAMPLES` section should be extended to include one or two examples that use the return values somehow.

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
modules/files/unarchive.py

##### ANSIBLE VERSION

```
ansible 2.9.2
  config file = None
  configured module search path = ['/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib64/python3.6/site-packages/ansible
  executable location = /usr/lib/python-exec/python3.6/ansible
  python version = 3.6.9 (default, Oct  5 2019, 11:39:46) [GCC 8.3.0]
```

##### CONFIGURATION
not relevant

##### OS / ENVIRONMENT
not relevant

### PR DESCRIPTION
**Pull request by paultaiton**
##### SUMMARY

Added the "timeout" parameter that is passed to "fetch_file" utility function. This will allow specifying a value other than the default 10 seconds so that it doesn't fail for big files or slow connections.

I've also added documentation for all the return values, which fixes #67445

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

unarchive

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by aminvakil** [comment]
I don't know if this is going to be merged or not, but this needs a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) if it's going to. Something like [this](https://github.com/ansible/ansible/blob/devel/changelogs/fragments/70318-dnf-add-nobest-option.yml).

**Comment by paultaiton** [comment]
@aminvakil 
This is my first contribution so I was not aware of that. Thanks for letting me know, I've added a fragment that I think is appropriate.

File: changelogs/fragments/72530-unarchive-timeout.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+minor_changes:
+  - unarchive - added a timeout parameter to the unarchive module so
```
Comment (aminvakil) [inline changelogs/fragments/72530-unarchive-timeout.yml]:
```suggestion
  - unarchive - add ``timeout`` option to the unarchive module so
```

File: changelogs/fragments/72530-unarchive-timeout.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    that a value other than the default 10 seconds may be given for
+    fetching from remote web sources. (https://github.com/ansible/ansible/pull/72530)
+  - unarchive - added 'RETURN' documentation (https://github.com/ansible/ansible/issues/67445)
```
Comment (aminvakil) [inline changelogs/fragments/72530-unarchive-timeout.yml]:
```suggestion
  - unarchive - add ``RETURN`` documentation (https://github.com/ansible/ansible/issues/67445).
```

File: changelogs/fragments/72530-unarchive-timeout.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+minor_changes:
+  - unarchive - added a timeout parameter to the unarchive module so
+    that a value other than the default 10 seconds may be given for
+    fetching from remote web sources. (https://github.com/ansible/ansible/pull/72530)
```
Comment (aminvakil) [inline changelogs/fragments/72530-unarchive-timeout.yml]:
```suggestion
    fetching from remote web sources (https://github.com/ansible/ansible/pull/72530).
```

**Comment by paultaiton** [comment]
I applied your suggestions. Thank you @aminvakil

**Comment by aminvakil** [comment]
@paultaiton You're welcome!

Few other notes from a non-maintainer, you can safely ignore these:
1) `timeout` is quite confusing in `unarchive` module, maybe another name?
2) stating your name in module after adding an option and RESULT is not appropriate.
3) `timeout` is not an useful option, I mean what are the odds of waiting more than 10 seconds for a stable http connection to connect?

**Comment by sivel** [comment]
Thank you very much for your interest in Ansible.  Per previous decisions, I am going to close this as a duplicate of https://github.com/ansible/ansible/issues/70432

Please see https://github.com/ansible/ansible/issues/70432#issuecomment-653092892 for more reasoning.

Our recommendation is that you call `get_url` with the advanced parameters, and then call `unarchive` on the downloaded file.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**